### PR TITLE
Enforce ChangeDetectionStrategy.OnPush

### PR DESCRIPTION
**Pull request by xelaint**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
`@daffodil/design` and `@daffodil/daffio` components are not CD.onPush by default

Part of: #1885

## What is the new behavior?
`@daffodil/design` and `@daffodil/daffio` components are CD.onPush by default

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: apps/daffio/.eslintrc.js [hunk 1]
```diff
@@ -16,6 +16,9 @@ module.exports = {
         createDefaultProgram: true
       },
       rules: {
+        '@angular-eslint/prefer-on-push-component-change-detection': [
+          'error'
+        ],
```
Comment (griest024) [inline apps/daffio/.eslintrc.js:21]: Do you think this should be part of our published eslint config? I think so.

Comment (damienwebdev) [reply apps/daffio/.eslintrc.js:21]: Yes, absolutely. But, we have large number of components in the repo that need to be checked before we have support across Daffodil and therefore are able to add it to our global config.

Comment (griest024) [reply apps/daffio/.eslintrc.js:21]: @xelaint can you add a todo to remove this once this is part of the global config?

Comment (xelaint) [reply apps/daffio/.eslintrc.js:22]: Added

**Comment by damienwebdev** [comment]
I have rebase merged this into ` develop`.

**Pull request merged**